test_runner/regress/test_wal_restore.py

```diff
@@ -125,3 +127,55 @@ def test_wal_restore_initdb(
         )
         log.info(f"original lsn: {original_lsn}, restored lsn: {restored_lsn}")
         assert restored.safe_psql("select count(*) from t", user="cloud_admin") == [(300000,)]
+
+
+def test_wal_restore_http(
+    neon_env_builder: NeonEnvBuilder,
+    pg_bin: PgBin,
+    test_output_dir: Path,
+    port_distributor: PortDistributor,
+    base_dir: Path,
+    pg_distrib_dir: Path,
+):
+    env = neon_env_builder.init_start()
+    endpoint = env.endpoints.create_start("main")
+    endpoint.safe_psql("create table t as select generate_series(1,300000)")
+    tenant_id = env.initial_tenant
+    timeline_id = env.initial_timeline
+    Lsn(endpoint.safe_psql("SELECT pg_current_wal_flush_lsn()")[0][0])
```

Comment:
Why the insertion and (no-op ?!) conversion of the flush LSN? You're not using that anywhere.

Reply:
You are right that it's redundant.

As for the why, as in how did it happen, this is an artifact of `ruff --fix` not going far enough. basically, I copy pasted the test from above, so originally it looked like `original_lsn = Lsn(endpoint.safe_psql("SELECT pg_current_wal_flush_lsn()")[0][0])`. Then, I removed the use of the `original_lsn` variable. ruff sees that and the `--fix` command removes the assignment but keeps the function call for the potential side effects. Here there are none that we are interested in, so yeah it can just be removed.

Reply:
I have removed it.